File: app/src/main/java/com/divercity/app/data/entity/group/AuthorInfo.java
package com.divercity.app.data.entity.group;

import com.google.gson.annotations.SerializedName;

public class AuthorInfo{

	@SerializedName("avatar_medium")
	private String avatarMedium;

	@SerializedName("avatar_thumb")
	private String avatarThumb;

	@SerializedName("name")
	private String name;

	@SerializedName("nickname")
	private String nickname;

	@SerializedName("last_name")
	private String lastName;

	@SerializedName("id")
	private int id;

	@SerializedName("present")
	private boolean present;

	public void setAvatarMedium(String avatarMedium){
		this.avatarMedium = avatarMedium;
	}

	public String getAvatarMedium(){
		return avatarMedium;
	}

	public void setAvatarThumb(String avatarThumb){
		this.avatarThumb = avatarThumb;
	}

	public String getAvatarThumb(){
		return avatarThumb;
	}

	public void setName(String name){
		this.name = name;
	}

	public String getName(){
		return name;
	}

	public void setNickname(String nickname){
		this.nickname = nickname;
	}

	public String getNickname(){
		return nickname;
	}

	public void setLastName(String lastName){
		this.lastName = lastName;
	}

	public String getLastName(){
		return lastName;
	}

	public void setId(int id){
		this.id = id;
	}

	public int getId(){
		return id;
	}

	public void setPresent(boolean present){
		this.present = present;
	}

	public boolean isPresent(){
		return present;
	}

	@Override
 	public String toString(){
		return 
			"AuthorInfo{" + 
			"avatar_medium = '" + avatarMedium + '\'' + 
			",avatar_thumb = '" + avatarThumb + '\'' + 
			",name = '" + name + '\'' + 
			",nickname = '" + nickname + '\'' + 
			",last_name = '" + lastName + '\'' + 
			",id = '" + id + '\'' + 
			",present = '" + present + '\'' + 
			"}";
		}
}

File: app/src/main/java/com/divercity/android/data/entity/group/group/AuthorInfo.kt
package com.divercity.android.data.entity.group.group

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class AuthorInfo(

    @field:SerializedName("avatar_medium")
    val avatarMedium: String? = null,

    @field:SerializedName("avatar_thumb")
    val avatarThumb: String? = null,

    @field:SerializedName("name")
    val name: String? = null,

    @field:SerializedName("last_name")
    val lastName: String? = null,

    @field:SerializedName("id")
    val id: Int? = null,

    @field:SerializedName("present")
    val present: Boolean? = null

) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readValue(Int::class.java.classLoader) as? Int,
        parcel.readValue(Boolean::class.java.classLoader) as? Boolean
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(avatarMedium)
        parcel.writeString(avatarThumb)
        parcel.writeString(name)
        parcel.writeString(lastName)
        parcel.writeValue(id)
        parcel.writeValue(present)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<AuthorInfo> {
        override fun createFromParcel(parcel: Parcel): AuthorInfo {
            return AuthorInfo(parcel)
        }

        override fun newArray(size: Int): Array<AuthorInfo?> {
            return arrayOfNulls(size)
        }
    }
}